src/routes/+page.server.ts

```diff
@@ -1,16 +1,49 @@
 import type { PageServerLoad } from './$types';
-import type { ArticleEntry } from '$lib/types';
+
+import { D1CacheName, cacheGet, cachePut } from '$lib/storage';
+import type { ArticleEntry, CacheResponse } from '$lib/types';
 import { lastMonth } from '$lib/time';
 
+import { error } from '@sveltejs/kit';
+
 export const load: PageServerLoad = async ({ platform, url }) => {
-	const after = url.searchParams.get('after') ??  lastMonth.toISOString().split('T')[0];
+	if (platform === undefined) {
+		error(404, { message: 'Platform is undefined' });
+	}
+
+	const after = url.searchParams.get('after') ?? lastMonth.toISOString().split('T')[0];
+	const cache = await platform.caches.open(D1CacheName);
+
+	const cacheEntry = (await cacheGet(url.origin, cache)) as CacheResponse | undefined;
 
-	const resp = await platform?.env.DB.prepare(
+	if (cacheEntry && cacheEntry.after! <= after) {
+		return {
+			entries: cacheEntry.entries,
+			after,
+			cached: true
+		};
+	}
+
+	const response = await platform.env.DB.prepare(
 		'SELECT date, images_published, images_published_with_alt_text, categories FROM articles WHERE date > ?'
-	).bind(after).all();
+	)
+		.bind(after)
+		.all();
+
+	if (response.error) {
+		error(400, { message: response.error });
+	}
+
+	platform.context.waitUntil(
+		cachePut(url.origin, cache, {
+			entries: response.results as ArticleEntry[] | [],
+			after
+		})
+	);
 
 	return {
-		entries: resp?.results as Array<ArticleEntry> | [],
-		after 
+		entries: response.results as ArticleEntry[] | [],
```

Comment:
I think this carried over from a while ago, but does TypeScript get upset if we simplify `ArticleEntry[] | []` to `ArticleEntry[]`?

Reply:
Yeah, it'll say like Record<string, unknown>[] doesn't sufficiently overlap with ArticleEntry[]

Reply:
Sad